ibis_substrait/compiler/translate.py

```diff
@@ -937,12 +937,29 @@ def _contains(
     compiler: SubstraitCompiler,
     **kwargs: Any,
 ) -> stalg.Expression:
-    return stalg.Expression(
-        singular_or_list=stalg.Expression.SingularOrList(
-            value=translate(op.value, compiler, **kwargs),
-            options=[translate(value, compiler, **kwargs) for value in op.options],
+    if not isinstance(op.options, (collections.abc.Sequence, ir.ListExpr)):
+        # Column expression
+        return stalg.Expression(
+            subquery=stalg.Expression.Subquery(
+                in_predicate=stalg.Expression.Subquery.InPredicate(
+                    needles=[translate(op.value, compiler, **kwargs)],
+                    haystack=translate(
+                        op.options.to_projection(),
+                        compiler,
+                        # if we to a `to_projection` do we need to mess
+                        # with field offsets?
```

Comment:
I don't think so, but we'll have to handle outer references for correlated subqueries at some point. We can do that as a concerted effort in a follow-up PR.

Reply:
Ok -- does it make sense that we wouldn't pass through the `**kwargs` here?  Because if we do, 3 of the TPC queries fail because they are passing non-zero child offsets to `ops.Selection`

```
AssertionError: non-empty child_rel_field_offsets passed in to selection translation rule
```

where the `child_rel_field_offsets` in the `kwargs` are:
```
(Pdb) p child_rel_field_offsets
{UnboundTable(schema=ibis.Schema {
  ps_partkey     int32
  ps_suppkey     int32
  ps_availqty    int32
  ps_supplycost  decimal(prec=15, scale=2)
  ps_comment     string
}, name='partsupp'): 0, UnboundTable(schema=ibis.Schema {
  p_partkey      int32
  p_name         string
  p_mfgr         string
  p_brand        string
  p_type         string
  p_size         int32
  p_container    string
  p_retailprice  decimal(prec=15, scale=2)
  p_comment      string
}, name='part'): 5}
```